server/src/main/java/edu/jhu/server/Game.java

```diff
@@ -480,23 +485,32 @@ private void handleJoinRequest(JSONObject request, Player author) {
     joinNotification.put(Constants.PLAYER_TAG, author.getTag());
     joinNotification.put(Constants.PLAYER_SUSPECT, author.getSuspect().toString());
     notifyPlayers(joinNotification);
-  	
-  	// Once we reach 3 players, we can start the game. So start a 5 minute
-  	//	timer!
-  	if (this.players.size() == 3 && timer == null) {
-  		timer = new Timer();
-  		
-  		// In 5 minutes, start the game.
-  		timer.schedule(new TimerTask() {
-			    @Override
-			    public void run() {
-			        start();
-			    }
-			}, Constants.START_GAME_AFTER_MS);
-  	}
+
+    // Once we reach 3 players, we can start the game. So start a 5 minute
+    // timer!
+    if (this.players.size() == 3 && timer == null) {
+      timer = new Timer();
+
+      timeToStart = Constants.START_GAME_AFTER_MS;
+      timer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          if (timeToStart <= 0) {
+            start();
+          } else {
+            JSONObject notification = new JSONObject();
```

Comment:
I actually made a function to handle one line notifications like this called handleSimpleRequest(player, EventType), but it would require iteratating over players, so no biggie.